File: archetype-base/src/test/java/com/alo7/archetype/http/HttpUtilsTest.java
// Copyright 2019 Alo7 Inc. All rights reserved.

package com.alo7.archetype.http;

import org.junit.Assert;
import org.junit.Test;

import java.util.HashMap;
import java.util.Map;

public class HttpUtilsTest {
    @Test
    public void concatPathTest() {
        Assert.assertEquals(HttpUtils.concatPath("xxx.com/", "/test"), "xxx.com/test");
        Assert.assertEquals(HttpUtils.concatPath("xxx.com/", "test"), "xxx.com/test");
        Assert.assertEquals(HttpUtils.concatPath("xxx.com", "/test"), "xxx.com/test");
        Assert.assertEquals(HttpUtils.concatPath("xxx.com", "test"), "xxx.com/test");
    }

    @Test
    public void formatUrlWithPathParamsTest() {
        Map<String, Object> map = new HashMap<>();
        map.put("path1", 1);
        map.put("path2", 2);
        Assert.assertEquals(HttpUtils.formatUrlWithPathParams("xxx.com/api/1/test", map), "xxx.com/api/1/test");
        Assert.assertEquals(HttpUtils.formatUrlWithPathParams("xxx.com/api/{path1}/test", map), "xxx.com/api/1/test");
        Assert.assertEquals(HttpUtils.formatUrlWithPathParams("xxx.com/api/{path1}/{path2}", map), "xxx.com/api/1/2");
    }

    @Test
    public void formatUrlWithParamsTest() {
        Map<String, Object> map = new HashMap<>();
        map.put("key1", 1);
        map.put("key2", 2);
        Assert.assertEquals(HttpUtils.formatUrlWithParams("xxx.com/api/1/test", map),
                "xxx.com/api/1/test?key1=1&key2=2");
        Assert.assertEquals(HttpUtils.formatUrlWithParams("xxx.com/api/1/test?key0=0", map),
                "xxx.com/api/1/test?key0=0&key1=1&key2=2");
    }
}

File: archetype-common/src/test/java/com/kelin/archetype/common/http/HttpUtilsTest.kt
// Copyright 2020 Kelin Inc. All rights reserved.

package com.kelin.archetype.common.http

import com.kelin.archetype.common.http.HttpUtils.concatPath
import com.kelin.archetype.common.http.HttpUtils.formatUrlWithParams
import com.kelin.archetype.common.http.HttpUtils.formatUrlWithPathParams
import com.kelin.archetype.test.KtTestUtils
import org.junit.Test

/**
 * @author Kelin Tan
 */
class HttpUtilsTest : KtTestUtils {
    @Test
    fun concatPathTest() {
        concatPath("xxx.com/", "/test") eq "xxx.com/test"
        concatPath("xxx.com/", "test") eq "xxx.com/test"
        concatPath("xxx.com", "/test") eq "xxx.com/test"
        concatPath("xxx.com", "test") eq "xxx.com/test"
    }

    @Test
    fun formatUrlWithPathParamsTest() {
        val map = mapOf("path1" to 1, "path2" to 2)
        formatUrlWithPathParams("xxx.com/api/1/test", map) eq "xxx.com/api/1/test"
        formatUrlWithPathParams("xxx.com/api/{path1}/test", map) eq "xxx.com/api/1/test"
        formatUrlWithPathParams("xxx.com/api/{path1}/{path2}", map) eq "xxx.com/api/1/2"
    }

    @Test
    fun formatUrlWithParamsTest() {
        val map = mapOf("key1" to 1, "key2" to 2)
        formatUrlWithParams("xxx.com/api/1/test", map) eq "xxx.com/api/1/test?key1=1&key2=2"
        formatUrlWithParams("xxx.com/api/1/test?key0=0", map) eq "xxx.com/api/1/test?key0=0&key1=1&key2=2"
    }
}